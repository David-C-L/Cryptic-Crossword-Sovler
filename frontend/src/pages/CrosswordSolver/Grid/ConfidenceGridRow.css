.row {
    height: 2em;
    width: 100%;
}

.grid-row {
    border: 0;
    height: 2em;
    margin: 0;
    padding: 0;
}
