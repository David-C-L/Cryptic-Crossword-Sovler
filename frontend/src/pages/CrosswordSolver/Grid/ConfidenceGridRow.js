import ConfidenceGridEntry from "./ConfidenceGridEntry";
import './ConfidenceGridRow.css'

function ConfidenceGridRow(props) {
    return (
        <tr className="grid-row row" key={props.row}>
            {
                props.values && props.values.map((value, col) =>
                    <ConfidenceGridEntry row={props.row} col={col} value={value} updateCell={props.updateCell} editable={props.editable} confidence={props.confidence[col]} lastFilled={props.lastFilled ? props.lastFilled[col] : false} showConfidence={props.showConfidence} clueNo={props.cluesPositions ? props.cluesPositions[col] : false} workingOn={props.workingOn ? props.workingOn[col] : false}/>
                )
            }
        </tr>
    )
}

export default ConfidenceGridRow;