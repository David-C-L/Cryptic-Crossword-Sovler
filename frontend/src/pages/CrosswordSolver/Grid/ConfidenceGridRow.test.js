import React from "react";
import { unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';
import ConfidenceGridRow from "./ConfidenceGridRow";
import { BrowserRouter as Router } from "react-router-dom";


let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Router><ConfidenceGridRow /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});