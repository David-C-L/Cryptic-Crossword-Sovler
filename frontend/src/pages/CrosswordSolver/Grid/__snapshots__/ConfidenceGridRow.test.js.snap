// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  className="grid-row row"
/>
`;
